src/hotspot/share/prims/jvmtiTagMapTable.cpp

```diff
@@ -113,6 +113,10 @@ JvmtiTagMapEntry* JvmtiTagMapTable::find(int index, unsigned int hash, oop obj)
 }
 
 JvmtiTagMapEntry* JvmtiTagMapTable::find(oop obj) {
+  if (obj->has_no_hash()) {
```

Comment:
This new function you added checks if the markWord has a hashCode. If there is a displaced markWord, then it very well might be that there is a hashCode, but it is in the displaced markWord - either in a stack lock or an ObjectMonitor. Bailing here does not seem correct, as it might actually be in the table even if there is no hashCode in the markWord. Is this an optimization?

Reply:
It is an optimization.  I don't think we want to create an identity hash for all oops just for lookup.  Is there a better way to find if an oop hashCode?

Reply:
I was hoping there was just a bit but you're right.  I renamed it to fast_no_hash_check() and only return true if the object is unlocked and added a comment.

Reply:
I'm rerunning jvmti and jdi tests locally.